gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -443,15 +440,9 @@ gst_tensor_src_merge_tensor_by_type (GstTensorInfo * info, guint size)
     }
   }
 
-  /** Now merge into 1 tensor using the outermost dimension*/
-  for (dim_idx = NNS_TENSOR_RANK_LIMIT - 1; dim_idx > 0; dim_idx--) {
-    if (info[base_idx].dimension[dim_idx] != 1) {
-      dim_idx += 1;
-      break;
-    }
-  }
+  /** Now merge into 1 tensor using the innermost dimension*/
```

Comment:
Are you merging n x 4:1:4:1 to 4:n:4:1?

Reply:
I am merging n x 4:1:4:1 to 4:n:4:1 (4 x dimension[] = {4, 1, 4, 1} -> dimension[] = {4, 4, 4, 1})
Please correct me if i am wrong - my understanding is dimension[0] is the innermost dimension.

Reply:
Could you please share the rationale why you are merging innermost rank with dimension 1? It appears to be counter-intuitive.

Reply:
My primary rationale was - if the innermost dimension remains 1, then it is just adding unnecessary single dimension in the array. Further, merging using the innermost rank makes outer ranks available to future merging/aggregating/formatting of the tensor.

Merging can be done with the outer dimension (n x 1:4:1:1 -> 1:4:n:1) with a squeeze function (1:4:n:1 -> 4:n:1:1). Please let me know if this is more appropriate and I will update?

Reply:
Use decides the dimension. Thus if there is an inner dimension with 1 whose outer dimension is > 1, it is intentional (or we should assume so). Unless there are additional properties dictating which dimension to be merged, I'd suggest to use the innermost outer-dimension of > 1 dimension; e.g.. n x 1:4:1:1 --> 1:4:n:1).

For the flexibility, optionally accepting merging-dimension (rank) looks appropriate as well.

Reply:
Updated. Added functionality to merge tensor with innermost/outermost/innermost-outer dimension.

For the IIO data, the data will use dimension `Num of Channels:Buffer Capacity:1:1`.